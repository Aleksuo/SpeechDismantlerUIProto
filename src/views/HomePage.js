import React from 'react'
import { Button, Paper, Grid, Fab} from '@material-ui/core'
import MicIcon from '@material-ui/icons/Mic'
import PauseIcon from '@material-ui/icons/Pause'
import PropTypes from 'prop-types'

const Timer = ({elapsed}) =>{
	const elapsedSec = Math.round(elapsed/1000)
	const min = Math.floor(elapsedSec/60)
	const sec = elapsedSec-(min*60)
	return(
		<div>
			<h2>{min}:{sec}</h2>
		</div>
	)
}


const Transcript = ({ transcript }) => {
	const items = transcript.map((word, idx) => { return <span key={idx}>{word.word} </span> })
	return (
		<div>
			<Paper elevation={3} style={{ maxHeight: "30vh", height: "30vh", overflow: "auto" }}>
				{items}
			</Paper>
		</div>
	)
}


Transcript.propTypes = {
	transcript: PropTypes.array
}

const Interim = ({ interim }) => {
	return (
		<div>
			<Paper elevation={2} style={{ color: "gray", height: "5vh", textAlign: 'center', }}>
				{interim}
			</Paper>
		</div>
	)
}



Interim.propTypes = {
	interim: PropTypes.string
}



const HomePage = (props) =>{ 
    const {state, toggleRecord, reset} = props

    return(
            <div>
					<Grid container
						spacing={24}
						direction="column"
						alignItems="center"
						justify="flex-end"
					>
						<Grid item xs={12}>
							<Fab aria-label="mic" color={state.isRecording ? 'secondary' : 'primary'} onClick={toggleRecord}>
								{state.isRecording ? <PauseIcon /> : <MicIcon />}
							</Fab>
						</Grid>
						<Grid item xs={12}>
							<Timer elapsed={state.elapsed}/>
						</Grid>
						<Grid item xs={6} md={3} style={{ width: "100%", height: "100%" }}>
							<Interim interim={state.interim} />
						</Grid>
						<Grid item xs={12} md={6} style={{ width: "100%", height: "100%" }}>
							<Transcript transcript={state.transcript} />
						</Grid>
						<Grid item xs={12}>
							<Button variant="contained" onClick={reset}>Reset</Button>
						</Grid>
					</Grid>
				</div>)
}

export default HomePage