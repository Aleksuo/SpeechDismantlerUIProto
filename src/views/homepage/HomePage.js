import React from 'react'
import { Button, Grid, Fab} from '@material-ui/core'
import MicIcon from '@material-ui/icons/Mic'
import PauseIcon from '@material-ui/icons/Pause'
import PropTypes from 'prop-types'

import Interim from './Interim'
import Transcript from './Transcript'
import Timer from './Timer'

const HomePage = (props) => {
    const { state, toggleRecord, reset } = props
    return (
        <div>
            <Grid container
                spacing={24}
                direction="column"
                alignItems="center"
                justify="flex-end"
            >
                <Grid item xs={12}>
                    <Fab aria-label="mic" color={state.isRecording ? 'secondary' : 'primary'} onClick={toggleRecord}>
                        {state.isRecording ? <PauseIcon /> : <MicIcon />}
                    </Fab>
                </Grid>
                <Grid item xs={12}>
                    <Timer elapsed={state.elapsed} />
                </Grid>
                <Grid item xs={6} md={3} style={{ width: "100%", height: "100%" }}>
                    <Interim interim={state.interim} />
                </Grid>
                <Grid item xs={12} md={6} style={{ width: "100%", height: "100%" }}>
                    <Transcript transcript={state.transcript} wordCounter = {state.wordCounter}/>
                </Grid>
                <Grid item xs={12}>
                    <Button variant="contained" onClick={reset}>Reset</Button>
                </Grid>
            </Grid>
        </div>)
}

HomePage.propTypes = {
    state: PropTypes.object,
    toggleRecord: PropTypes.func,
    reset: PropTypes.func
}

export default HomePage