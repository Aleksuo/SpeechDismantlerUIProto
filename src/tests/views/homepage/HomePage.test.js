/* eslint-disable */
import React from "react"
import Enzyme, {shallow, mount} from "enzyme"
import HomePage from "../../../views/HomePage/HomePage"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({adapter: new Adapter()})

const defaultState = {
	isRecording: false,
	elapsed: 0,
	transcript: [],
	interim: "",
	left: false,
}

describe("HomePage component", () =>{
    test("renders", () =>{
        const wrapper = shallow(<HomePage state={defaultState}/>)
        expect(wrapper).toMatchSnapshot()
    })
})