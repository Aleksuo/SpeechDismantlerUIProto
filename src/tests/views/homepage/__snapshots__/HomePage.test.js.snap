// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage
    state={
      Object {
        "elapsed": 0,
        "interim": "",
        "isRecording": false,
        "left": false,
        "transcript": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Grid)
        alignItems="center"
        container={true}
        direction="column"
        justify="flex-end"
        spacing={24}
      >
        <WithStyles(Grid)
          item={true}
          xs={12}
        >
          <WithStyles(Fab)
            aria-label="mic"
            color="primary"
          >
            <pure(MicIcon) />
          </WithStyles(Fab)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          xs={12}
        >
          <Timer
            elapsed={0}
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={3}
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
          xs={6}
        >
          <Interim
            interim=""
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={6}
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
          xs={12}
        >
          <Transcript
            transcript={Array []}
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          xs={12}
        >
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            Reset
          </WithStyles(Button)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": Array [
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(Fab)
              aria-label="mic"
              color="primary"
            >
              <pure(MicIcon) />
            </WithStyles(Fab)>
          </WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <Timer
              elapsed={0}
            />
          </WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            md={3}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
            xs={6}
          >
            <Interim
              interim=""
            />
          </WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            md={6}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
            xs={12}
          >
            <Transcript
              transcript={Array []}
            />
          </WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(Button)
              color="secondary"
              onClick={[Function]}
              variant="contained"
            >
              Reset
            </WithStyles(Button)>
          </WithStyles(Grid)>,
        ],
        "container": true,
        "direction": "column",
        "justify": "flex-end",
        "spacing": 24,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Fab)
              aria-label="mic"
              color="primary"
            >
              <pure(MicIcon) />
            </WithStyles(Fab)>,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "mic",
              "children": <pure(MicIcon) />,
              "color": "primary",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Timer
              elapsed={0}
            />,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "elapsed": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Interim
              interim=""
            />,
            "item": true,
            "md": 3,
            "style": Object {
              "height": "100%",
              "width": "100%",
            },
            "xs": 6,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "interim": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Transcript
              transcript={Array []}
            />,
            "item": true,
            "md": 6,
            "style": Object {
              "height": "100%",
              "width": "100%",
            },
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "blobUrl": undefined,
              "transcript": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="secondary"
              onClick={[Function]}
              variant="contained"
            >
              Reset
            </WithStyles(Button)>,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Reset",
              "color": "secondary",
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Reset",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Grid)
          alignItems="center"
          container={true}
          direction="column"
          justify="flex-end"
          spacing={24}
        >
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(Fab)
              aria-label="mic"
              color="primary"
            >
              <pure(MicIcon) />
            </WithStyles(Fab)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <Timer
              elapsed={0}
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            md={3}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
            xs={6}
          >
            <Interim
              interim=""
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            md={6}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
            xs={12}
          >
            <Transcript
              transcript={Array []}
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(Button)
              color="secondary"
              onClick={[Function]}
              variant="contained"
            >
              Reset
            </WithStyles(Button)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alignItems": "center",
          "children": Array [
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Fab)
                aria-label="mic"
                color="primary"
              >
                <pure(MicIcon) />
              </WithStyles(Fab)>
            </WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <Timer
                elapsed={0}
              />
            </WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              md={3}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              xs={6}
            >
              <Interim
                interim=""
              />
            </WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              md={6}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              xs={12}
            >
              <Transcript
                transcript={Array []}
              />
            </WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Button)
                color="secondary"
                onClick={[Function]}
                variant="contained"
              >
                Reset
              </WithStyles(Button)>
            </WithStyles(Grid)>,
          ],
          "container": true,
          "direction": "column",
          "justify": "flex-end",
          "spacing": 24,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Fab)
                aria-label="mic"
                color="primary"
              >
                <pure(MicIcon) />
              </WithStyles(Fab)>,
              "item": true,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "mic",
                "children": <pure(MicIcon) />,
                "color": "primary",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Timer
                elapsed={0}
              />,
              "item": true,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "elapsed": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Interim
                interim=""
              />,
              "item": true,
              "md": 3,
              "style": Object {
                "height": "100%",
                "width": "100%",
              },
              "xs": 6,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "interim": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Transcript
                transcript={Array []}
              />,
              "item": true,
              "md": 6,
              "style": Object {
                "height": "100%",
                "width": "100%",
              },
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "blobUrl": undefined,
                "transcript": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="secondary"
                onClick={[Function]}
                variant="contained"
              >
                Reset
              </WithStyles(Button)>,
              "item": true,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Reset",
                "color": "secondary",
                "onClick": [Function],
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Reset",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
